python/test/unit/language/test_core_amd.py

```diff
@@ -2212,6 +2220,52 @@ def test_convert2d(dtype, shape, src_layout, dst_layout, device='cuda'):
     assert torch.equal(z, x)
 
 
+@pytest.mark.parametrize("shape", [(64, 64)])
+@pytest.mark.parametrize("dtype", ['float16'])
+@pytest.mark.parametrize("src_layout", [MfmaLayout(warps_per_cta=[2, 1]), MfmaLayout(warps_per_cta=[4, 1])])
+@pytest.mark.parametrize("dst_layout", [BlockedLayout([1, 4], [4, 16], [1, 1], [1, 0])])
+def test_make_range(dtype, shape, src_layout, dst_layout, device='cuda'):
```

Comment:
test_make_range is not part of upstream testing. They use test_convert2d as a way of testing both conversions and make_range. However, ttgir in these tests uses load instruction in src layout, and loading with mfma layout is not supported at the moment. I created this test with similar IR that uses conversions before load instruction.

Reply:
Am I right, that there is no way to write a `@triton.jit` code that will produce needed IR?

Perhaps, another approach is not to run the IR on HW but add a `lit` test to check the correct IR processing (e.g. to LLVM IR or even ISA) ?

Reply:
You are right. From the compiler point of view, there is no need for mfma layout to be created unless there is a dot operation. For testing in such cases, ttgir is created directly and run.
For now I feel it is safer to have it run and make sure we produce the correct tensor, but I agree with you that
testing can be enhanced by checking produced llvm ir (or ISA) as well.